unlock-app/src/components/interface/locks/Settings/forms/UpdateHooksForm.tsx

```diff
@@ -157,61 +213,46 @@ export const UpdateHooksForm = ({
     }
   }
 
-  const toggleField = (field: FormPropsKey) => {
-    const fieldStatus = enabledFields[field]
-    setEnabledFields({
-      ...enabledFields,
-      [field]: !fieldStatus,
-    })
-
-    if (fieldStatus) {
-      setValue(field, DEFAULT_USER_ACCOUNT_ADDRESS)
-    }
-  }
-
   const disabledInput =
     disabled || setEventsHooksMutation.isLoading || isLoading
 
   return (
-    <form className="grid gap-6" onSubmit={handleSubmit(onSubmit)}>
-      {Object.entries(HookMapping)?.map(
-        ([field, { label, fromPublicLockVersion = 0, hookName }]) => {
-          const fieldName = field as FormPropsKey
-          const hasRequiredVersion = version && version >= fromPublicLockVersion
-          const enabled: boolean = enabledFields[fieldName] ?? false
-          const hasError = errors?.[hookName as FormPropsKey]
-
-          return (
-            hasRequiredVersion && (
-              <div key={hookName}>
-                <ToggleSwitch
-                  title={label}
-                  enabled={enabled}
-                  setEnabled={() => toggleField(fieldName)}
-                  disabled={disabledInput}
-                />
-                <Input
-                  {...register(fieldName, {
-                    validate: isValidAddress,
-                  })}
-                  disabled={disabledInput || !enabled}
-                  placeholder="Contract address, for ex: 0x00000000000000000"
-                  error={hasError && 'Enter a valid address'}
-                />
-              </div>
+    <FormProvider {...methods}>
+      <form
+        className="grid gap-6"
+        onSubmit={methods.handleSubmit(onSubmit)}
+        onChange={() => {
+          methods.trigger()
+        }}
+      >
+        {Object.entries(HookMapping)?.map(
+          ([field, { label, fromPublicLockVersion = 0, hookName }]) => {
+            const fieldName = field as FormPropsKey
+            const hasRequiredVersion =
+              version && version >= fromPublicLockVersion
+
+            if (!hasRequiredVersion) return null
+
+            return (
+              <HookSelect
+                key={hookName}
+                label={label}
+                name={fieldName}
+                disabled={disabledInput}
+              />
```

Comment:
This is here that you will show `CustomContractHook` based on the value selected in `HookSelect`